ldes-server-domain/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/domain/ldes/retentionpolicy/creation/RetentionPolicyCreatorImpl.java

```diff
@@ -0,0 +1,32 @@
+package be.vlaanderen.informatievlaanderen.ldes.server.domain.ldes.retentionpolicy.creation;
+
+import be.vlaanderen.informatievlaanderen.ldes.server.domain.ldes.retentionpolicy.RetentionPolicy;
+import be.vlaanderen.informatievlaanderen.ldes.server.domain.ldes.retentionpolicy.timebased.TimeBasedRetentionPolicy;
+import be.vlaanderen.informatievlaanderen.ldes.server.domain.viewcreation.valueobjects.RetentionConfig;
+import be.vlaanderen.informatievlaanderen.ldes.server.domain.viewcreation.valueobjects.ViewSpecification;
+
+import java.util.List;
+
+import static be.vlaanderen.informatievlaanderen.ldes.server.domain.ldes.retentionpolicy.timebased.TimebasedRetentionProperties.DURATION;
+
+public class RetentionPolicyCreatorImpl implements RetentionPolicyCreator {
+
+	@Override
+	public List<RetentionPolicy> createRetentionPolicyListForView(ViewSpecification viewSpecification) {
+		return viewSpecification
+				.getRetentionConfigs()
+				.stream()
+				.map(this::getRetentionPolicy)
+				.toList();
+	}
+
+	private RetentionPolicy getRetentionPolicy(RetentionConfig retentionConfig) {
+		// TODO update so that multiple retention policies can easily be incorporated
+		if ("timebased".equals(retentionConfig.getName())) {
```

Comment:
Magic string? :)

Reply:
Haha, that's true. From now on a free sandwich for every magic string in the code.
In my defense, it was already in the code :-)